sphinx_doc_src/fish_sphinxext.py

```diff
@@ -0,0 +1,82 @@
+import os.path
+import re
+
+from docutils import nodes
+from docutils.parsers import rst
+from docutils.statemachine import ViewList
+
+STRING_DOC = os.path.join(os.path.dirname(__file__), "cmds/string.rst")
+with open(STRING_DOC) as fd:
+    STRING_DOC_LINES = list(fd)
+
+
+def string_doc_lines(start, stop):
+    """Print lines from the string documentation start from the line that matches start
+    and stopping tat the first line that matches stop."""
+    result = []
+    i = 0
+    while i < len(STRING_DOC_LINES):
+        line = STRING_DOC_LINES[i]
+        if not result:
+            if re.match(start, line):
+                i += 1
+                result += [line, STRING_DOC_LINES[i]]  # always include the header's ---
```

Comment:
Shouldn't you be able to get by without the `i += 1` here if you just add the first line?

That would also then allow you to rewrite it as a `for` loop which is more idiomatic here.

Something like

```python
    result = []
    for line in STRING_DOC_LINES:
        if not result:
            if re.match(start, line):
                result += [line]
        elif re.match(stop, line):
                return result
        else:
            result += [line]
```

Reply:
This is a hack - for the subcommand description I want to start at a header line and stop at a line with all hyphens - but not the one immediately after the header. We should not need this logic if we restructure the doc sources.